modules/vault-cluster/main.tf

```diff
@@ -64,7 +64,8 @@ resource "google_compute_instance_template" "vault_public" {
   }
 
   network_interface {
-    network = "${var.network_name}"
+    network = "${var.subnetwork_name != "" ? "" : var.network_name}"
```

Comment:
I'm a little confused by this line. Why do we want `network` set to `var.network_name` only when `var.subnetwork_name` is non-empty?

Reply:
Most likely:
> Either network **or** subnetwork must be provided.

[Docs](https://www.terraform.io/docs/providers/google/r/compute_instance.html#network)

Reply:
@nimerix has got it!

You can also see the same thing in the `consul-cluster` module of the `terraform-google-consul` project [here](https://github.com/hashicorp/terraform-google-consul/blob/master/modules/consul-cluster/main.tf#L65).